pandas/tests/io/test_html.py

```diff
@@ -113,6 +113,28 @@ def spam_data(self, datapath):
     def banklist_data(self, datapath):
         return datapath("io", "data", "html", "banklist.html")
 
+    @pytest.fixture
+    def gh_13141_data(self):
+        return """
+          <table>
+            <tr>
+              <th>HTTP</th>
+              <th>FTP</th>
+              <th><a href="https://en.wiktionary.org/wiki/linkless">Linkless</a></th>
+            </tr>
+            <tr>
+              <td><a href="https://en.wikipedia.org/">Wikipedia</a></td>
+              <td><a href="ftp://ftp.us.debian.org/">Debian</a></td>
+              <td>Linkless</td>
+            </tr>
+            <tfoot>
+              <tr>
+                <td><a href="https://en.wikipedia.org/wiki/Page_footer">Footer</a></td>
+              </tr>
+            </tfoot>
+          </table>
+          """
+
```

Comment:
```suggestion
    @pytest.mark.fixture
    def gh_13141_expected(self):
        return {
            "head_ignore": ["HTTP", "FTP", "Linkless"],
            "head_extract": [("HTTP", None), ("FTP", None), ("Linkless", "https://en.wiktionary.org/wiki/linkless")],
            "body_ignore": ["Wikipedia", "Debian", "Linkless"],
            "body_extract": [("Wikipedia", "https://en.wikipedia.org/"), ("Debian", "ftp://ftp.us.debian.org/"), ("Linkless", None)],
            "footer_ignore": ["Footer", None, None],
            "footer_extract": [("Footer", "https://en.wikipedia.org/wiki/Page_footer"), (None, None), (None, None)]
        }

```
Note I was not sure if your return type on extract was `None` or `(None, None)` when `extract_links=True` but nothing is available to parse.
Not sure which is best, either.

Reply:
`(None, None)` currently. I went with this so that they are all consistent and commands like `df.str[0]` don't fail.